contracts/contracts/harvest/BaseHarvester.sol

```diff
@@ -58,38 +83,46 @@ abstract contract BaseHarvester is Governable {
     /**
      * Address receiving rewards proceeds. Initially the Vault contract later will possibly
      * be replaced by another contract that eases out rewards distribution.
-     */
+     **/
     address public rewardProceedsAddress;
 
     /**
-     * @dev Constructor to set up initial internal state
-     * @param _vaultAddress Address of the Vault
-     */
-    constructor(address _vaultAddress) {
-        require(address(_vaultAddress) != address(0));
+     * All tokens are swapped to this token before it gets transferred
+     * to the `rewardProceedsAddress`. USDT for OUSD and WETH for OETH.
+     **/
+    address public immutable baseTokenAddress;
+    uint256 public immutable baseTokenDecimals;
+
+    mapping(address => address[]) public uniswapV2Path;
+    mapping(address => bytes) public uniswapV3Path;
+    mapping(address => bytes32) public balancerPoolId;
+
+    struct CurvePoolData {
+        // TODO: Check if storing as single slot would help with gas here
```

Comment:
It would help with gas.

Reply:
Have changed it to pack it into a single slot